src/services/api/credentials.ts

```diff
@@ -111,7 +153,361 @@ export class Credentials {
+		if (!entity && providerId) {
+			entity = await this.repository.findOne({
+				where: {
+					providerCredentialId: id,
+					providerId: providerId,
+					customer: { customerId: customer.customerId },
+				},
+				relations: ['customer'],
+			});
+		}
+
+		// Strategy 3: Search across all provider fields
+		if (!entity) {
+			entity = await this.repository.findOne({
+				where: {
+					providerCredentialId: id,
+					customer: { customerId: customer.customerId },
+				},
+				relations: ['customer'],
+			});
+		}
+		if (!entity) {
+			return null;
+		}
+
+		// Sync status from provider if requested
+		if (syncStatus) {
+			await this.syncStatusFromProvider(entity);
+			// Reload entity from database to get updated status
+			const updatedEntity = await this.repository.findOne({
+				where: {
+					issuedCredentialId: entity.issuedCredentialId,
+				},
+				relations: ['customer'],
+			});
+			if (updatedEntity) {
+				entity = updatedEntity;
+			}
+		}
+
+		// Optionally include full credential
+		let credential: VerifiableCredential | null = null;
+		if (includeCredential) {
+			credential = await this.fetchCredentialFromProvider(entity);
+		}
+		return this.toResponse(entity, { includeCredential, credential: credential ?? undefined });
+	}
+	/**
+	 * Create a new issued credential record
+	 */
+	async create(data: {
+		providerId: string;
+		providerCredentialId?: string;
+		issuerId?: string;
+		subjectId?: string;
+		format: 'jwt' | 'jsonld' | 'sd-jwt-vc' | 'anoncreds';
+		type: string[];
+		status?: 'issued' | 'suspended' | 'revoked';
+		statusUpdatedAt?: Date;
+		issuedAt: Date;
+		expiresAt?: Date;
+		credentialStatus?: Record<string, any>;
+		metadata?: Record<string, any>;
+		customer: CustomerEntity;
+	}): Promise<IssuedCredentialEntity> {
+		const entity = this.repository.create({
+			providerId: data.providerId,
+			providerCredentialId: data.providerCredentialId,
+			issuerId: data.issuerId,
+			subjectId: data.subjectId,
+			format: data.format,
+			type: data.type,
+			status: data.status || 'issued',
+			statusUpdatedAt: data.statusUpdatedAt,
+			issuedAt: data.issuedAt,
+			expiresAt: data.expiresAt,
+			credentialStatus: data.credentialStatus,
+			metadata: data.metadata,
+			customer: data.customer,
+		});
+
+		return await this.repository.save(entity);
+	}
+
+	/**
+	 * Update credential status
+	 */
+	async updateStatus(
+		issuedCredentialId: string,
+		status: 'issued' | 'suspended' | 'revoked',
+		customer: CustomerEntity
+	): Promise<boolean> {
+		const result = await this.repository.update(
+			{
+				issuedCredentialId,
+				customer: { customerId: customer.customerId },
+			},
+			{
+				status,
+				statusUpdatedAt: new Date(),
+			}
+		);
+
+		return (result.affected ?? 0) > 0;
+	}
+	/**
+	 * Check if string is a valid UUID
+	 */
+	private isUUID(str: string): boolean {
+		const uuidRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
+		return uuidRegex.test(str);
+	}
```

Comment:
Validations can be enacted natively; use `uuid`'s embedded format and version methods documented here: https://www.npmjs.com/package/uuid#uuidvalidatestr.